python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -317,11 +427,20 @@ def categorical_indices_from_vocabulary_file(feature_info, feature_tensor, file_
         vocabulary_keys: values of the vocabulary stated in the vocabulary_file.
     """
     vocabulary_keys, vocabulary_ids = get_vocabulary_info(feature_info, file_io)
-    num_oov_buckets = feature_info.get("feature_layer_info")["args"].get("num_oov_buckets", 1)
+    args = feature_info.get("feature_layer_info")["args"]
+
+    if "dropout_rate" in args:
+        default_value = 0  # Default OOV index when using dropout
+        num_oov_buckets = None
```

Comment:
@lastmansleeping It concerns me to see a default value defined here (and for `num_oov_buckets`) instead of in `parse_args`, but is the reasoning because we want to customize it depending on whether we're using a dropout rate? I guess I'll be easy to convince if you can explain a little bit ;)

Reply:
Yes. You are right. You choose which function you want to use from the feature config as opposed to the command line arguments.